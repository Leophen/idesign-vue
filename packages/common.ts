import _ from 'lodash'
import {
  Component,
  getCurrentInstance,
  Slots,
  VNode,
} from 'vue'

// 通用位置类型
export type placementType =
  | 'top'
  | 'top-left'
  | 'top-right'
  | 'bottom'
  | 'bottom-left'
  | 'bottom-right'
  | 'left'
  | 'left-top'
  | 'left-bottom'
  | 'right'
  | 'right-top'
  | 'right-bottom'

// 通用触发类型
export type triggerType = 'hover' | 'click' | 'context-menu'

// 转换 width/height/left/... 样式值
export const turnValue = (val: string | number | undefined) => {
  if (!val) {
    return
  }
  if (_.isNumber(val as number)) {
    return `${val}px`
  }
  return val
}

// const _stateValue = ref(defaultValue)
// const innerValue = computed(() => value ?? _stateValue.value)

/**
 * 渲染 default slot，获取子组件 VNode。处理多种子组件创建场景
 * @returns {function(childComponentName: string, slots: Slots): VNode[]}
 * @param childComponentName
 * @param slots
 * @example const getChildByName = useChildComponentSlots()
 * @example getChildComponentByName('xxItem')
 */
export const useChildComponentSlots = () => {
  const instance = getCurrentInstance()
  return (childComponentName: string, slots?: Slots): VNode[] => {
    if (!slots) {
      slots = instance?.slots
    }
    const content = slots?.default?.() || []

    return (
      content
        .map((item: VNode) => {
          if (item.children && Array.isArray(item.children))
            return item.children
          return item
        })
        .flat()
        // @ts-ignore
        .filter((item: VNode) =>
          (item.type as Component).name?.endsWith(childComponentName)
        ) as VNode[]
    )
  }
}
