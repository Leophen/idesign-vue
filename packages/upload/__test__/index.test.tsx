import { Upload } from '../index';
import { render } from '../../common'

describe('Upload 组件测试', () => {
  it('create', () => {
    const wrapper = render(<Upload />);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('block', () => {
    const wrapper = render(<Upload theme="block" />);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('drag', () => {
    const wrapper = render(<Upload theme="drag" />);
    expect(wrapper.element).toMatchSnapshot();
  });
});

