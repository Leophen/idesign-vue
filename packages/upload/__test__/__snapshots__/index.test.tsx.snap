// Vitest Snapshot v1

exports[`Upload 组件测试 > block 1`] = `
<div
  class="i-upload"
>
  <input
    style="display: none;"
    type="file"
  />
  
  <!--v-if-->
  <div
    class="i-upload__block"
  >
    <i
      class="i-icon icon-ThePlus"
      style="font-size: 20px;"
    />
     点击上传
  </div>
  <!--v-if-->
  
</div>
`;

exports[`Upload 组件测试 > create 1`] = `
<div
  class="i-upload"
>
  <input
    style="display: none;"
    type="file"
  />
  
  <button
    class="i-button i-button--type-info i-button--variant-outline"
  >
    <i
      class="i-icon icon-ArrowLineUpload i-button-icon"
    />
    
    点击上传
    
  </button>
  <!--v-if-->
  <!--v-if-->
  
</div>
`;

exports[`Upload 组件测试 > drag 1`] = `
<div
  class="i-upload"
>
  <input
    style="display: none;"
    type="file"
  />
  
  <!--v-if-->
  <!--v-if-->
  <div
    class="i-upload__drag"
    draggable=""
  >
    <i
      class="i-icon icon-ArrowLineUpload"
      style="font-size: 20px;"
    />
    <span
      class="i-upload-placeholder"
    >
      点击或拖动到框内上传
    </span>
    <span
      class="i-upload-tip"
    >
      支持不超过 10M 的 excel 类型文件
    </span>
  </div>
  
</div>
`;
